frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -40,6 +40,12 @@ end
 function ReaderWikipedia:registerKeyEvents()
     if Device:hasKeyboard() then
         self.key_events.ShowWikipediaLookup = { { "Alt", "W" }, { "Ctrl", "W" } }
+        if Device.model == "Kindle3" then
+            -- Override previous shortcut on kindle3 as the kernel designates Alt+W as "2".
+            -- Any user-facing documentation should _still_ advertise this shortcut as Alt+W, as
+            -- that is what the user must input in order to use it.
+            self.key_events.ShowWikipediaLookup = { { "2" } }
+        end
```

Comment:
Wondering if this shouldn't be put into Device, so we don't have to copy/paste that comment everywhere we'll need Alt+QWERTY... ? Something like:
```lua
if Device.model == "Kindle3" then
  self.key_events.ShowWikipediaLookup = { Device.CrapMapAlt["W"] }
end
```

Reply:
What's this crap map supposed to look like?

Reply:
Something similar to the `local sym_key_map` you added recently:
```lua
Device.CrapMapAlt = {
 ["W"] = { "2" }, ...
}
```
(With a name that would read well and natural when used.)

Dunno what your Krapdle returns with Shift+Alt+W. If it returns Shift + 2, I guess it could work.
If not, it would need a `Device.getCrapMapAlt = function() ... end` returning all the `{ { ... } }` combinations to have it work.
@NiLuJe : thoughts?

Reply:
I don't rightly recall exactly how the Key stuff works exactly, but, sure, centralizing this sounds pretty good to me ;).

Reply:
like this @poire-z 

```lua
-- Kindle's kernel designates Alt+'QWERTYUIOP' as numerical values on kindle 3.
-- This crapMap is intended to make shortcut writting more readable, for example Alt+W
-- would be: self.key_events = { Device.CrapMapAlt["W"] } instead of self.key_events = { "2" }
local CrapMapAlt = {
    ["Q"] = {"1"}, ["W"] = {"2"},
    ["E"] = {"3"}, ["R"] = {"4"},
    ["T"] = {"5"}, ["Y"] = {"6"},
    ["U"] = {"7"}, ["I"] = {"8"},
    ["O"] = {"9"}, ["P"] = {"0"},
}
```

Reply:
Yep, with a more politically correct name.

Reply:
> Yep, with a more politically correct name.

deep down, I knew you don't have what it takes to write that in the code. 

where about is that suppose to go anyway? where in Device?

Reply:
deep down :)

I guess it could be a property of the Kindle3 device. Not a function returning true or false like the other, but a present/nil value (so we don't need to let other Devices know about that crap), so instead of
```diff
-if Device.model == "Kindle3" then
+if Device.crapAltMap then
```
But I'll let @NiLuJe tell us, he does have what it takes :)

Reply:
Possibly a dedicated table/file like the other layouts in `frontend/device/kindle/`, with a name like 3.x_alt_transform or something

Reply:
Then cache that from kindle/device in the relevant device blocks (which, err, is just the K3 I guess? so perhaps `k3_` will be less jargony that `3.x_` as the prefix ;).

Reply:
<img src="https://github.com/user-attachments/assets/1e4e65e1-8462-4447-b5f2-0102d8a770b4" width=30%>

Reply:
I am going to be looking the other way for some unspecified amount of time, if anyone wants to perform the slight of hand whilst I am busy looking at butterflies, don't mention it afterwards ;)

Reply:
okay I've created the file, but I am a bit confused about the rest, because this isn't a normal event map, but a pretend one that simply relabels things... how would one go about caching that in Device?